x/stake/types.go

```diff
@@ -0,0 +1,186 @@
+package stake
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	crypto "github.com/tendermint/go-crypto"
+)
+
+// Params defines the high level settings for staking
+type Params struct {
+	InflationRateChange sdk.Rat `json:"inflation_rate_change"` // maximum annual change in inflation rate
+	InflationMax        sdk.Rat `json:"inflation_max"`         // maximum inflation rate
+	InflationMin        sdk.Rat `json:"inflation_min"`         // minimum inflation rate
+	GoalBonded          sdk.Rat `json:"goal_bonded"`           // Goal of percent bonded atoms
+
+	MaxValidators uint16 `json:"max_validators"` // maximum number of validators
+	BondDenom     string `json:"bond_denom"`     // bondable coin denomination
+}
+
+func defaultParams() Params {
+	return Params{
+		InflationRateChange: sdk.NewRat(13, 100),
+		InflationMax:        sdk.NewRat(20, 100),
+		InflationMin:        sdk.NewRat(7, 100),
+		GoalBonded:          sdk.NewRat(67, 100),
+		MaxValidators:       100,
+		BondDenom:           "fermion",
+	}
+}
+
+//_________________________________________________________________________
+
+// Pool - dynamic parameters of the current state
+type Pool struct {
+	TotalSupply       int64   `json:"total_supply"`        // total supply of all tokens
+	BondedShares      sdk.Rat `json:"bonded_shares"`       // sum of all shares distributed for the Bonded Pool
+	UnbondedShares    sdk.Rat `json:"unbonded_shares"`     // sum of all shares distributed for the Unbonded Pool
+	BondedPool        int64   `json:"bonded_pool"`         // reserve of bonded tokens
+	UnbondedPool      int64   `json:"unbonded_pool"`       // reserve of unbonded tokens held with candidates
+	InflationLastTime int64   `json:"inflation_last_time"` // block which the last inflation was processed // TODO make time
+	Inflation         sdk.Rat `json:"inflation"`           // current annual inflation rate
+}
+
+// XXX define globalstate interface?
+
+func initialPool() Pool {
+	return Pool{
+		TotalSupply:       0,
+		BondedShares:      sdk.ZeroRat,
+		UnbondedShares:    sdk.ZeroRat,
+		BondedPool:        0,
+		UnbondedPool:      0,
+		InflationLastTime: 0,
+		Inflation:         sdk.NewRat(7, 100),
+	}
+}
+
```

Comment:
Desirable to load default staking params / pool params from a config file? That way clients can customize without recompiling.

Reply:
same as previous response above ^